Выполнил задание 2

pagaaev opened the pull request.

Commit:
Выполнил задание 2

okryukova commented on .allure/allure-2.15.0/bin/allure:
⛔️Нужно исправить.  Папку .allure не нужно было загружать в репозиторий. Эта папка должна быть добавлена в .gitignore.

okryukova commented on .idea/encodings.xml:
⛔️Нужно исправить.  Папку .idea не нужно было загружать в репозиторий. Эта папка должна быть добавлена в .gitignore.

okryukova commented on Readme:
⚠️Можно улучшить. Хорошей практикой является добавление README.md с описанием проекта, которое включает название, технологии в проекте (например, Java 11, JUnit 4.13.2, maven 3.9.0 и т.д.), настройку (если необходима) и как запускать (mvn clean test - кавычки позволяют выполнять запуск из файла README.md). Для оформления данных в виде таблицы используй сочетание клавиш Ctrl + N или Cmd + N. Как оформлять README.md можно ознакомиться в статье https://blog.jetbrains.com/idea/2022/08/markdown-support-improvements/

okryukova commented on src/main/java/org/example/Ingredient.java:
```diff
@@ -0,0 +1,87 @@
+package org.example;
+
+public class Ingredient {
+    private String _id;
```
⛔️Нужно исправить.  Неверный нейминг. Спец символы в названии полей не используются

okryukova commented on src/main/java/org/example/Main.java:
⛔️Нужно исправить.  Класс не используется. Удали его, пожалуйста

okryukova commented on src/main/java/org/example/User.java:
```diff
@@ -0,0 +1,37 @@
+package org.example;
+
+public class User {
```
⚠️Можно улучшить. Можно использовать @DaTa и @AllArgsConstructor из библиотеки Lombok

okryukova commented on src/test/java/CreateOrderTest.java:
```diff
@@ -0,0 +1,81 @@
+    }
+
+    @Test
+    @DisplayName("Check status code of create order")
```
⚠️Можно улучшить. В allure есть еще одна аннотация для описания теста Description. Добавь ее, пожалуйста, во всех тестах проекта

okryukova commented on src/test/java/CreateOrderTest.java:
```diff
@@ -0,0 +1,81 @@
+        ingredientIds.add(ingredients.get(1).get_id());
+        ingredientIds.add(ingredients.get(2).get_id());
+        Response response = orderAPI.createOrder(ingredientIds, "");
+        response.then().assertThat().body("success", equalTo(true))
```
⛔️Нужно исправить.  Сначала проверяем код ответа, а затем его тело. Если код неверный, проверка тела не имеет смысла. Поправь, пожалуйста, во всем коде

okryukova commented on src/test/java/CreateOrderTest.java:
```diff
@@ -0,0 +1,81 @@
+        Response response = orderAPI.createOrder(ingredientIds, accessToken);
+        response.then().assertThat()
+                .statusCode(SC_INTERNAL_SERVER_ERROR);
+        userAPI.deleteUser(accessToken);
```
⛔️Нужно исправить. Шаг удаления юзера нужно вынести в метод с аннотацией after. Он должен выполняться даже если тест упал.  Поправь, пожалуйста, во всем коде

okryukova commented on src/test/java/CreateUserTest.java:
```diff
@@ -0,0 +1,78 @@
+
+    @Before
+    public void setUp() {
+        RestAssured.baseURI = "https://stellarburgers.nomoreparties.site";
```
⛔️Нужно исправить.  Base URL присваиваем один раз в проекте. Можно использовать request specification в RestAssured или утилитный базовый класс.

okryukova commented on src/test/java/CreateUserTest.java:
```diff
@@ -0,0 +1,78 @@
+        User user = UserUtils.getRandomUser();
+        user.setEmail(null);
+        Response createResponse = userAPI.createUser(user);
+        createResponse.then().assertThat().body("success", equalTo(false))
```
⛔️Нужно исправить. Для всех негативных тестов нужно также проверить сообщение об ошибке. Поправь, пожалуйста, во всем коде

okryukova commented on src/test/java/GetOrderTest.java:
```diff
@@ -0,0 +1,41 @@
+    @Test
+    @DisplayName("Check status code of get orders")
+    public void getOrdersTest() {
+        String accessToken = userAPI.createUser();
```
⛔️Нужно исправить.  Метод создания пользователя нужно вынести в метод с аннотацией before. Поправь, пожалуйста, во всем коде

okryukova commented on src/test/java/UserTest.java:
```diff
@@ -0,0 +1,83 @@
+                .statusCode(SC_UNAUTHORIZED);
+        userAPI.deleteUser(accessToken);
+    }
+}
```
⛔️Нужно исправить.  Нужно добавить аллюр отчет в пулл реквест

okryukova replied:
Не исправлено

okryukova replied:
Нет отчета

Commit:
Add .allure/ to .gitignore and remove from repo

Commit:
Add .idea/ to .gitignore and remove from repo

Commit:
Исправлен нейминг поля _id на id в классе Ingredient

Commit:
Fix test assertions order: check status code before body in CreateOrderTest

Commit:
Refactor CreateOrderTest: move user deletion to @After to ensure cleanup

Commit:
refactor: unify baseURI setup and move user deletion to @After

- RestAssured.baseURI задан один раз в статическом блоке, чтобы не повторять в каждом тесте
- Перенес удаление пользователя в @After метод, чтобы удаление выполнялось всегда, даже если тест упал
- Добавлены проверки сообщений об ошибках в негативных тестах createUser

Commit:
Refactor GetOrderTest: move user creation to @Before and user deletion to @After for better test isolation

Commit:
Добавлены Allure @Step аннотации для UserTest для улучшения отчёта

okryukova commented on src/main/java/org/example/Main.java:
⛔️Нужно исправить. Удали, пожалуйста, неиспользуемый код

okryukova commented on src/main/java/org/example/OrderAPI.java:
```diff
@@ -0,0 +1,52 @@
+        }.getType());
+    }
+
+    public Response createOrder(List<String> ingredientIds, String accessToken) {
```
⛔️Нужно исправить. Для всех методов, описывающих шаги теста, необходимо использовать аннотацию @Step

okryukova commented on src/test/java/CreateOrderTest.java:
```diff
@@ -0,0 +1,92 @@
+
+    @Before
+    public void setUp() {
+        RestAssured.baseURI = "https://stellarburgers.nomoreparties.site";
```
⛔️Нужно исправить.  Base URL присваиваем один раз в проекте. Можно использовать request specification в RestAssured или утилитный базовый класс.

okryukova commented on src/test/java/CreateOrderTest.java:
```diff
@@ -0,0 +1,92 @@
+        response.then().assertThat()
+                .statusCode(SC_BAD_REQUEST)
+                .and()
+                .body("success", equalTo(false));
```
⛔️Нужно исправить. Не проверен текст ошибке  в теле ответа

okryukova commented on src/test/java/GetOrderTest.java:
```diff
@@ -0,0 +1,50 @@
+        Response response = orderAPI.getOrders("");
+        response.then().assertThat().body("success", equalTo(false))
+                .and()
+                .statusCode(SC_UNAUTHORIZED);
```
⛔️Нужно исправить. Не проверен текст ошибки в теле ответа

okryukova commented on src/test/java/LoginUserTest.java:
```diff
@@ -0,0 +1,58 @@
+        Response loginResponse = userAPI.loginUser(user);
+        loginResponse.then().assertThat().body("success", equalTo(false))
+                .and()
+                .statusCode(SC_UNAUTHORIZED);
```
⛔️Нужно исправить. Аналогично, нет проверки на текст ошибки

okryukova commented on src/test/java/UserTest.java:
```diff
@@ -0,0 +1,98 @@
+                .and()
+                .statusCode(SC_OK);
+
+        accessToken = loginResponse.path("accessToken");
```
⛔️Нужно исправить. Шаг удаления нужно вынести в метод с аннотацией after

okryukova commented on src/test/java/UserTest.java:
```diff
@@ -0,0 +1,98 @@
+        Response changeResponse = changeUser(firstUser, firstUserAccessToken);
+        changeResponse.then().assertThat().body("success", equalTo(false))
+                .and()
+                .statusCode(SC_FORBIDDEN);
```
⛔️Нужно исправить. Проверь, пожалуйста, сообщение об ошибке в теле

okryukova commented on src/test/java/UserTest.java:
```diff
@@ -0,0 +1,98 @@
+        Response changeResponse = changeUser(changedUser, "");
+        changeResponse.then().assertThat().body("success", equalTo(false))
+                .and()
+                .statusCode(SC_UNAUTHORIZED);
```
⛔️Нужно исправить. Тут тоже проверяем тело ответа

Commit:
Удалён неиспользуемый Main.java по требованию ревью

Commit:
Добавлены @Step для методов OrderAPI (Allure)

Commit:
Вынесен baseURI в BaseTest, добавлены изменения во все классы

Commit:
Добавлена проверка текста ошибки в тесте createOrderWithoutIngredientsTest

Commit:
Добавлена проверка текста ошибки при получении заказов без авторизации, наследование от BaseTest

Commit:
Добавлена проверка текста ошибки при неуспешном логине в LoginUserTest

Commit:
Добавлена проверка текста ошибки при изменении пользователя с уже существующим email в UserTest

Commit:
Исправлены проверки тела ответа и рефакторинг UserTest согласно замечаниям

okryukova commented on src/test/java/GetOrderTest.java:
```diff
@@ -0,0 +1,53 @@
+        Response response = orderAPI.getOrders("");
+        response.then().assertThat()
+                .body("success", equalTo(false))
+                .body("message", equalTo("You should be authorised")) // Проверяем текст ошибки
```
⛔️Нужно исправить.  С начала проверяем код ответа, затем тело. Если код неверный, то проверять тело не нужно
